### PR TITLE
Spiders that spawn from player-laid eggs are now AI controlled.

### PR DESCRIPTION
**Pull request by BurgerLUA**
## About The Pull Request

Spiders that spawn from player-laid eggs are now AI controlled, meaning that they are no longer ALSO player controlled.

## Why It's Good For The Game

Let's compare spiders to xenos in terms of balance when controlled by players.

| Point | Spiders | Xenomorphs |
| --- | --- | --- |
Breeding Method | Lay an egg in a monkey or a human, regardless of health, and you'll get ~10 spiderlings in a few minutes. | Lay a single egg in a conscious human to get one larvae. |
| Childhood | Scatter into vents at the speed of bullshit. Takes 3 hits with an object to kill. Turn into an adult instantly without warning. Since there is so many of you, there will ALWAYS be adults. | Sit inside a conscious human hoping they don't escape their bed. Pop out of them when it's time and then wait another x minutes to evolve while anything more powerful than a fart can kill you.
Speed | Lightning fucking fast. | Lightning fucking fast. |
Health | 200 god damn health | 125 health. |
Melee Damage | 15 to 20. Deals 60 damage to objects. | 25. Deals 60 damage to objects. |
Player Obtainment | Xenobiology Gold Core Slime RNG and Sentient Transfer Potion. Takes about 30-45 minutes to get if you know what you're doing. | Xenomorph Alien Hive raid + Surgery. Takes about 20-30 minutes to clear a dangerous Xeno Hive. Another 10-15 minutes to find and have someone do surgery for you to insert ALL the organs.
| Weaknesses | Lack of Air. | Fire. |

image

Spiders are literally xenomorphs except better. Their numbers are absolutely insane compared to xenomorphs and that is because of this fundamental fact: Spiders are balanced around being controlled by an AI and not players who override their movement and attack speed. Antags can easily create situations by throwing a few corpses to the spiders and it then making a nest and making more spiders; that is the purpose of spiders. The balance did not anticipate that a player would preemptively hoard the maximum amount of monkeys and then turn themselves into a spider to then lay eggs in all the monkeys to create a near infinite stream of player-controlled spiders as long as there were salty players looking to murderbone.

You can still create bullshit spider armies, however they are not player controlled and do not have the benefits of being player controlled, which means that spiders are now balanced as they should be.

## Changelog
:cl: BurgerBB
balance: Spiders can no longer be ghost controlled if they're laid by players.
/:cl:

### CODE REVIEW

**Comment by Fractious** [comment]
Spider infestation is a gimmick xenobiologists can pull off, and it never happens most of the time. Sure the spiders have a lot of HP's, but they're slow as shit and can be lasered super easily. Maybe nerf the vent crawl speed, but they're in a fine spot imo.

**Comment by r4d6** [comment]
> Spider infestation is a gimmick xenobiologists can pull off, and it never happens most of the time. Sure the spiders have a lot of HP's, but they're slow as shit and can be lasered super easily. Maybe nerf the vent crawl speed, but they're in a fine spot imo.

You were not in that round, I was the warden and I could not hit a spide dragging a fridge behind it. Since it blocked my shot and was too agile for me to shoot when exposed.

**Comment by Soranayo** [comment]
> 
> 
> > Spider infestation is a gimmick xenobiologists can pull off, and it never happens most of the time. Sure the spiders have a lot of HP's, but they're slow as shit and can be lasered super easily. Maybe nerf the vent crawl speed, but they're in a fine spot imo.
> 
> You were not in that round, I was the warden and I could not hit a spide dragging a fridge behind it. Since it blocked my shot and was too agile for me to shoot when exposed.

I was that spider, i recall even calling you 'Nerd' but i mean, you literally took no attempt at going around the fridge. At this point it was more of a you issue. (Also my goal was to stall you while the others did some work, if i went head-to-head i would've died easily.)

As for the matter of the PR itself, well instead of removing the spider ghost control altogether, maybe alter the number of spiderling that hatches from the body to a lower count instead ? Also i wouldn't say Xenos are below spiders, one swarms, the other picks off key separated targets and bring them somewhere safe until hatching, then swarms when they know they have the advantage, xenos are overall harder to kill because of the speed, stun, neurotox and instant-sleeps from the huggers.

TL;DR : Just keep the player control but lower the spiderling count, we rarely see gimmicks and honestly, i don't think removing them will help when it comes to people complaining that antags always use the same shidding. (Edit : I'm sleepy, ree)

**Comment by Poojawa** [comment]
:salt: 🍔

**Comment by BurgerLUA** [comment]
>>salt

excellent criticism

**Comment by deathride58** [comment]
This is an extremely heavyhanded nerf that doesn't actually resolve the one single issue that makes spiders broken as fuck.

The issue with spiders is that they're supposed to be a roundending mob exclusive to changelings, but they're available via gold cores due to an oversight. I'd be fine with removing their ability to spawn with gold cores, but this PR is just blatantly driven via salt

**Comment by BurgerLUA** [comment]
>salt

PR isn't driven by salt but sure lets throw that word around. like mature adults.

Spiders being from gold cores isn't the issue it's how powerful player controlled spiders are. They are exceptionally powerful and I don't see how that's a better PR than this current one. Antags having their own spider army is fine but having it so that all spiders that come from it be player is the issue.

**Comment by Soranayo** [comment]
> 
> 
> > salt
> 
> PR isn't driven by salt but sure lets throw that word around. like mature adults.
> 
> Spiders being from gold cores isn't the issue it's how powerful player controlled spiders are. They are exceptionally powerful and I don't see how that's a better PR than this current one. Antags having their own spider army is fine but having it so that all spiders that come from it be player is the issue.

Powerful is the least thing i would qualify the spiders to be as. It takes around 10 hits to kill someone, you can't block their way, and you're slower than them. Meanwhile a few good hits with a ranged weapon and you go down. Spider's issue is the swarming imo, not the player control.

**Comment by BurgerLUA** [comment]
> > > salt
> > 
> > 
> > PR isn't driven by salt but sure lets throw that word around. like mature adults.
> > Spiders being from gold cores isn't the issue it's how powerful player controlled spiders are. They are exceptionally powerful and I don't see how that's a better PR than this current one. Antags having their own spider army is fine but having it so that all spiders that come from it be player is the issue.
> 
> Powerful is the least thing i would qualify the spiders to be as. It takes around 10 hits to kill someone, you can't block their way, and you're slower than them. Meanwhile a few good hits with a ranged weapon and you go down. Spider's issue is the swarming imo, not the player control.

I don't think you read anything in the OP. It does not take 10 hits to kill.

**Comment by Soranayo** [comment]
> 
> 
> > > > salt
> > > 
> > > 
> > > PR isn't driven by salt but sure lets throw that word around. like mature adults.
> > > Spiders being from gold cores isn't the issue it's how powerful player controlled spiders are. They are exceptionally powerful and I don't see how that's a better PR than this current one. Antags having their own spider army is fine but having it so that all spiders that come from it be player is the issue.
> > 
> > 
> > Powerful is the least thing i would qualify the spiders to be as. It takes around 10 hits to kill someone, you can't block their way, and you're slower than them. Meanwhile a few good hits with a ranged weapon and you go down. Spider's issue is the swarming imo, not the player control.
> 
> I don't think you read anything in the OP. It does not take 10 hits to kill.

Oh no, i did read it, it was more of an hyperbole. If i recall correctly, the damage depends on the spider type, same goes for the amount of toxins injected per hit (I'm not sure about that, so hey, don't quote me.) but even then, since they have zero Armor pen, and the moment sentient spiders are announced, everyone grabs some melee armor, i can assure you it does take at least five to six seconds of uninterrupted hits to down someone if not more, and that's if they don't move, the toxin is a non-factor for quick fights since it's slow to metabolize and easy to get rid of. You die much quicker than you kill as a spider, so my point still stands.

On another note, jumping directly to the removal of a feature because of -one- bad round literally five minutes after the said round ended makes me understand why people think it's a salt PR. I'm proposing a nerf, instead of directly removing it, killing gimmicks will just lead to the same stuff, the same stuff that people already complain about.

**Comment by BurgerLUA** [comment]
>Oh no, i did read it, it was more of an hyperbole. If i recall correctly, the damage depends on the spider type, same goes for the amount of toxins injected per hit (I'm not sure about that, so hey, don't quote me.) but even then, since they have zero Armor pen, and the moment sentient spiders are announced, everyone grabs some melee armor, i can assure you it does take at least five to six seconds of uninterrupted hits to down someone if not more, and that's if they don't move, the toxin is a non-factor for quick fights since it's slow to metabolize and easy to get rid of. You die much quicker than you kill as a spider, so my point still stands.

This is citadel. Teamwork rarely happens. The round turned into a clusterfuck because the Head of Security and the Captain rushed into to deal head on with the spider threat and died. Crew wasn't armed or anything and no one understood how serious the threat was because everyone thought it was a greenshift meme.


>On another note, jumping directly to the removal of a feature because of -one- bad round literally five minutes after the said round ended makes me understand why people think it's a salt PR. I'm proposing a nerf, instead of directly removing it, killing gimmicks will just lead to the same stuff, the same stuff that people already complain about.

Time should be absolutely irrelevant when it comes to discussing the quality of the PR. Would people like me to autistically create 1 week reminders that tell me to make a PR about a feature? The only difference this would make is the conversation would be held 1 week later and context would be out of people's minds.

**Comment by Soranayo** [comment]
> 
> 
> This is citadel. Teamwork rarely happens. The round turned into a clusterfuck because the Head of Security and the Captain rushed into to deal head on with the spider threat and died. Crew wasn't armed or anything and no one understood how serious the threat was because everyone thought it was a greenshift meme.
> 
Irrelevant. If sec has no clue how to deal with a morph should we nerf the morph ? Of course not. If the clockies steamroll on the crew because they have no clue how to counter should we nerf them ? Nope. Ignorance is a non-factor when it comes to mechanic balancing. It's a player issue, not a game one. (Edit : Even when it comes to the lack teamwork, you should not have too much issue fighting spiders considering the slow speed.)
> 
> Time should be absolutely irrelevant when it comes to discussing the quality of the PR. Would people like me to autistically create 1 week reminders that tell me to make a PR about a feature? The only difference this would make is the conversation would be held 1 week later and context would be out of people's minds.

I was referring to why people thought it was a salt PR, not that it was one, nor was i talking about the quality of the PR, that'd be ridiculous from me since i have no clue how to code. I'm discussing balance possibilities and change, if i am for or against, not the PR quality.

**Comment by chefferz** [comment]
Make less spiderlings spawn from eggs.

Literally bam, the issue is done.

The only problem is that spiders can snowball STUPID fast when left unchecked, which is exactly what happened in the round that made you do this PR. Nobody checked in on xenobiology for over 20 minutes while a literal ton of spider eggs were hatching in them, which allowed it to snowball this fast.

**Comment by Ludox235** [comment]
I ded PR

Fun fucking fact xenomorphs larvas still bursts even if the host died after impregnation, oops!

**Comment by lolman360** [comment]
> Fun fucking fact xenomorphs larvas still bursts even if the host died after impregnation, oops!

intended

**Comment by DiscordWizard** [comment]
Well damn there was a big discussion here. Why not just make less spiders hatch from the eggs, they are seriously less powerful than xemomorph when adult. They are slower and they can't instantly decimate your stamina, the only thing they got over the aliens is poison which every single doctor with two brain cells can deal with and the breeding rate which should definitely be lowered

**Comment by lolman360** [comment]
>spidertaur does spider gimmick

ban he

**Comment by Soranayo** [comment]
> 
> 
> Well damn there was a big discussion here. Why not just make less spiders hatch from the eggs, they are seriously less powerful than xemomorph when adult. They are slower and they can't instantly decimate your stamina, the only thing they got over the aliens is poison which every single doctor with two brain cells can deal with and the breeding rate which should definitely be lowered

Exactly what i've been saying.

**Comment by deathride58** [comment]
Since this PR is obsolete as a result of other PRs, I'll be closing this one.
